### PR TITLE
doc: improves doc of new Query API

### PR DESCRIPTION
**Pull request by monperrus**
improves the doc. after merging in your PR, this will be ready for reaching master.

### CODE REVIEW

File: doc/filter.md [hunk 1]
```diff
@@ -93,22 +40,43 @@ list3 = rootPackage.filterChildren(
...
+```java
+// collecting all methods of deprecated classes
+list2 = rootPackage
+    .filterChildren(new AnnotationFilter(Deprecated.class))
+    .filterChildren(new TypeFilter(CtMethod.class)).list();
+```
+
+`CtQueryable#map(CtFunction)`enables you to give Java 8 lambda as query.
+A boolean return value of the lambda tells whether the elements
+are selected for inclusion or input not.
```
Comment (pvojtechovsky) [inline doc/filter.md]: There seems to be a mistake

File: doc/filter.md [hunk 2]
```diff
@@ -93,22 +40,43 @@ list3 = rootPackage.filterChildren(
...
-list4 = rootPackage
-  .filterChildren((CtMethod method)->method.getAnnotation(Deprecated.class)!=null)
-  .map(new OverriddenMethodFilter())
-  .list();
-```
\ No newline at end of file
+  .list()
+;
+```
+
```
Comment (pvojtechovsky) [inline doc/filter.md:81]: If CtFunction returns Iterable or an Array then each item is sent to next query step or result.

File: doc/filter.md (hunk 1 above)
Comment (monperrus) [reply doc/filter.md]: what do you mean?

File: doc/filter.md (hunk 2 above)
Comment (monperrus) [reply doc/filter.md:81]: added. does it actually work, is it tested?

**Comment by pvojtechovsky** [comment]
Thanks for fixing documentation! Your English and explanations are better then mine.
I have added some comments. I do not persist on changing the doc if it was intentional. Just let me know that it is done and I will merge it.

File: doc/filter.md (hunk 1 above)
Comment (pvojtechovsky) [reply doc/filter.md]: "or input not"

Comment (pvojtechovsky) [reply doc/filter.md]: may be there should be 
"are selected for inclusion of input or not"

File: doc/filter.md (hunk 2 above)
Comment (pvojtechovsky) [reply doc/filter.md:81]: the code like this
map((CtClass clazz)->clazz.getMethods())
tests it. I did not checked how much tests are there after your simplification. I can have a look tommorrow. I plan to finish today soon.

**Comment by monperrus** [comment]
done. you can merge.

thanks a lot for this great idea of chained queries and Java8 lambdas.

I look forward to the scope-inference query design.

**Comment by monperrus** [comment]
does it mean that the size of the final list is the sum of all
intermediate lists?

**Comment by pvojtechovsky** [comment]
The size of the final list is collection of all values returned from the last query step only.

**Pull request merged**